modules/apps/organizations/organizations-test/src/testIntegration/java/com/liferay/organizations/object/system/test/OrganizationSystemObjectDefinitionManagerTest.java

```diff
@@ -0,0 +1,299 @@
+			).name(
+				"comments"
+			).system(
+				true
+			).build(),
+			iterator.next());
+
+		Assert.assertTrue(iterator.hasNext());
+
+		_assertSystemObjectFields(
+			new TextObjectFieldBuilder(
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(
+					LanguageUtil.get(LocaleUtil.getDefault(), "name"))
+			).name(
+				"name"
+			).required(
+				true
+			).system(
+				true
+			).build(),
+			iterator.next());
+	}
+
+	@Test
+	public void testProperties() throws Exception {
+		Assert.assertEquals(
+			"L_ORGANIZATION",
+			_organizationSystemObjectDefinitionManager.
+				getExternalReferenceCode());
+		Assert.assertEquals(
+			LocalizedMapUtil.getLocalizedMap(
+				LanguageUtil.get(LocaleUtil.getDefault(), "organization")),
+			_organizationSystemObjectDefinitionManager.getLabelMap());
+		Assert.assertEquals(
+			LocalizedMapUtil.getLocalizedMap(
+				LanguageUtil.get(LocaleUtil.getDefault(), "organizations")),
+			_organizationSystemObjectDefinitionManager.getPluralLabelMap());
+		Assert.assertEquals(
+			ObjectDefinitionConstants.SCOPE_COMPANY,
+			_organizationSystemObjectDefinitionManager.getScope());
+		Assert.assertEquals(
+			"name",
+			_organizationSystemObjectDefinitionManager.
+				getTitleObjectFieldName());
+		Assert.assertEquals(
+			1, _organizationSystemObjectDefinitionManager.getVersion());
+	}
+
+	private long _addBaseModel(Map<String, Object> values) throws Exception {
+		return _organizationSystemObjectDefinitionManager.addBaseModel(
+			TestPropsValues.getUser(), values);
+	}
+
+	private void _assertCount(int count) throws Exception {
+		Assert.assertEquals(
+			count, _organizationLocalService.getOrganizationsCount());
+	}
+
+	private void _assertSystemObjectFields(
+		ObjectField expectedObjectField, ObjectField objectField) {
+
+		Assert.assertEquals(
+			expectedObjectField.getDBColumnName(),
+			objectField.getDBColumnName());
+		Assert.assertEquals(
+			expectedObjectField.getDBTableName(), objectField.getDBTableName());
+		Assert.assertEquals(
+			expectedObjectField.getDBType(), objectField.getDBType());
+		Assert.assertEquals(
+			expectedObjectField.isIndexed(), objectField.isIndexed());
+		Assert.assertEquals(
+			expectedObjectField.isIndexedAsKeyword(),
+			objectField.isIndexedAsKeyword());
+		Assert.assertEquals(
+			expectedObjectField.getIndexedLanguageId(),
+			objectField.getIndexedLanguageId());
+		Assert.assertEquals(
+			expectedObjectField.getLabelMap(), objectField.getLabelMap());
+		Assert.assertEquals(
+			expectedObjectField.getName(), objectField.getName());
+		Assert.assertEquals(
+			expectedObjectField.isRequired(), objectField.isRequired());
+		Assert.assertEquals(
+			expectedObjectField.isState(), objectField.isState());
+	}
+
+	@Inject
+	private ObjectDefinitionLocalService _objectDefinitionLocalService;
+
+	@Inject
+	private OrganizationLocalService _organizationLocalService;
+
+	private SystemObjectDefinitionManager
+		_organizationSystemObjectDefinitionManager;
```

Comment:
Would not be great to use `@Inject` here as well?
I am not sure, but I think we should use it since this is an OSGi component too.

Reply:
I followed the pattern of RelatedObjectEntryResourceTest. I think that since we need to instantiate in the method itself, we don't need the @inject.

Reply:
Great!